azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -40,65 +31,39 @@
 import java.nio.file.Paths;
 import java.nio.file.attribute.FileTime;
 import java.util.Optional;
-import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.zip.ZipFile;
-import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
-class FlowPreparer {
-
-  // Name of the file which keeps project directory size
-  static final String PROJECT_DIR_SIZE_FILE_NAME = "___azkaban_project_dir_size_in_bytes___";
+public class FlowPreparer extends AbstractFlowPreparer {
```

Comment:
Please add comments for the class and describe that it enables caching.

Reply:
will do